packages/contracts-bedrock/src/L2/SuperchainWETH.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { WETH98 } from "src/dispute/weth/WETH98.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { L1Block } from "src/L2/L1Block.sol";
+import { IL2ToL2CrossDomainMessenger } from "src/L2/IL2ToL2CrossDomainMessenger.sol";
+import { ETHLiquidity } from "src/L2/ETHLiquidity.sol";
+import { ISuperchainERC20Extensions } from "src/L2/ISuperchainERC20.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+
+import "src/libraries/errors/CommonErrors.sol";
+
+/// @title SuperchainWETH
+/// @notice SuperchainWETH is a version of WETH that can be freely transfered between chains within
+///         the superchain. SuperchainWETH can be converted into native ETH on chains that do not
+//          use a custom gas token.
+contract SuperchainWETH is WETH98, ISuperchainERC20Extensions, ISemver {
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0-beta.1";
+
+    /// @inheritdoc WETH98
+    function deposit() public payable override {
+        if (L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).isCustomGasToken()) revert NotCustomGasToken();
+        super.deposit();
+    }
+
+    /// @inheritdoc WETH98
+    function withdraw(uint256 wad) public override {
+        if (L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).isCustomGasToken()) revert NotCustomGasToken();
+        super.withdraw(wad);
+    }
+
+    /// @inheritdoc ISuperchainERC20Extensions
+    function sendERC20(address dst, uint256 wad, uint256 chainId) public {
+        // Burn from user's balance.
+        _burn(msg.sender, wad);
+
+        // Burn to ETHLiquidity contract.
+        if (!L1Block(Predeploys.L1_BLOCK_ATTRIBUTES).isCustomGasToken()) {
+            ETHLiquidity(Predeploys.ETH_LIQUIDITY).burn{ value: wad }();
+        }
```

Comment:
I think it's useful to avoid inflating the ETH supply arbitrarily -- keeping the extra supply inside of ETHLiquidity and treating it like a black hole seems easier for accounting purposes.

Reply:
When you say we could just keep ether in this contract, that involves active liquidity management on our side. The invariant is that there should never be liquidity constraints when sending ether between chains

Reply:
> When you say we could just keep ether in this contract, that involves active liquidity management on our side. The invariant is that there should never be liquidity constraints when sending ether between chains

There shouldn't be active management needed. The idea would be that when burning, the ETH you send stays in SuperchainWETH (instead of being forwarded to ETHLiquidity). When minting, if there isn't enough ETH in SuperchainWETH, it pulls from ETHLiquidity to get the rest. This conditional makes it a bit clunkier, and I agree the accounting is cleaner with the current design.

Reply:
@tynes Another question here—what is the rationale for the ETHLiquidity contract in the first place + [possibly lifting the restriction](https://github.com/ethereum-optimism/optimism/pull/11042#discussion_r1690590194) on who can call it in the future? It seems that we could also just seed SuperchainWETH directly with 2^248 wei, which lets us delete the ETHLiquidity contract, delete some logic in SuperchainWETH to mint/burn from it, and remove the extra selfdestruct/transfers

Reply:
Two reasons for ETHLiquidity
- More accurate `totalSupply` on superchain weth
- Very explicit access patterns, ie there isn't just a ton of `SELFBALANCE` in evm execution which makes me worried about bugs

Reply:
Got it. Also just want to confirm https://github.com/ethereum-optimism/optimism/pull/11042#discussion_r1692016112 and see what the use cases are here?

Reply:
The specific thing is flexibility, there are no known usecases as of right now but we could hardfork in other contracts that have the ability to source this liquidity

Now that the `DepositTx` type is supported in many places, we may not need to ever deprecate it, but you could imagine attempts to reduce diff in the future where deposits don't mint as part of the state transition function but instead they mint as part of EVM execution. Obviously this would be a big breaking change and i am not advocating for it